src/libopensc/pkcs15-esteid2018.c

```diff
@@ -61,6 +61,8 @@ static int sc_pkcs15emu_esteid2018_init(sc_pkcs15_card_t *p15card) {
 	for (j = 0; j < taglen; j++)
 		if (!isalnum(tag[j]))
 			LOG_FUNC_RETURN(card->ctx, SC_ERROR_INTERNAL);
+	if (p15card->tokeninfo->serial_number)
+		free(p15card->tokeninfo->serial_number);
```

Comment:
You can free safely null pointer, if check is redundant.

> The free() function shall cause the space pointed to by ptr to be deallocated; that is, made available for further allocation. If ptr is a null pointer, no action shall occur.

Reply:
Right. I did it this way as it was done similarly in all the other places, but I will try to adjust that, probably using `set_string` function, which is used in some of the drivers.